### PR TITLE
Basic working of Youtube search app.

### PR DESCRIPTION
**Pull request by saadyousafarbi**
This **PR** contains the basic working of the app with the help of **React** library. The user will be provided with following functionalities:

* User will be able to enter a `query` of what videos he/she want to watch,
* Relevant videos will be fetched from **Youtube** and matched videos will be displayed with the first video running as `default` video in a **Youtube video player**,
* Upon clicking on any other matched video, the default video will be replaced by the selected video in **Youtube video player**.

_Search results will change as user's input changes._

### CODE REVIEW

File: youtube-search-with-react/src/App.js [hunk 1]
```diff
@@ -1,27 +1,47 @@
...
-        </p>
-      </div>
-    );
-  }
+    constructor(props) {
+        super(props);
+        this.state = {
+            videos: [],
+            selectedVideo: null,
+            };
```
Comment (zubair-arbi) [inline youtube-search-with-react/src/App.js]: nit: dedent

File: youtube-search-with-react/src/App.js [hunk 2]
```diff
@@ -1,27 +1,47 @@
...
+        this.videoSearch('hello');
```
Comment (zubair-arbi) [inline youtube-search-with-react/src/App.js]: Lets remove this default search of videos with name/title `hello`. Instead we display a simple search field just like `google`.

File: youtube-search-with-react/src/App.js [hunk 3]
```diff
@@ -1,27 +1,47 @@
...
+            this.setState({
+                videos: videos,
+                selectedVideo: videos[0],
+            });
+        });
+    }
+
+    render() {
+        return (
+              <div>
```
Comment (zubair-arbi) [inline youtube-search-with-react/src/App.js]: nit: use proper indentation

File: youtube-search-with-react/src/App.js [hunk 4]
```diff
@@ -1,27 +1,47 @@
...
+            selectedVideo: null,
+            };
+        this.videoSearch('hello');
+    }
+
+    videoSearch(term) {
+        YTSearch({key: key, term: term}, (videos) => {
+            this.setState({
+                videos: videos,
+                selectedVideo: videos[0],
```
Comment (zubair-arbi) [inline youtube-search-with-react/src/App.js]: I think the default behavior should be to just display a list of videos and upon selection of a video by the use we should show video detail component.

**Review by zubair-arbi — APPROVED** [review]
Overall changes looks good ,just few small nits. Nice work 👍

**Comment by saadyousafarbi** [comment]
Squashed from 4d0c35530b90d1ae70da51bd5b4dedd252728f6c

**Pull request merged**